### PR TITLE
validate if member is over 18

### PR DESCRIPTION
**Pull request by AlexSJ1990**
check if member is under 18 - if so, don't allow them to sign up
improve validation of personnummer by checking that the first 8 digits make up a valid date

### CODE REVIEW

File: src/main/java/com/hedvig/botService/dataTypes/SSNSweden.java [hunk 1]
```diff
@@ -34,6 +35,17 @@ public boolean validate(String input) {
       this.errorMessage = "Personnummret måste skrivas med 12 siffor.";
       return false;
     }
+
+    try {
+      LocalDate.parse(
+        trimmedInput.substring(0, 4) + "-" + trimmedInput.substring(4, 6) + "-" + trimmedInput.substring(6, 8)
+      );
+    }
+    catch(Exception exception) {
```
Comment (palmenhq) [inline src/main/java/com/hedvig/botService/dataTypes/SSNSweden.java]: We should specify the exception we want to catch (`DateTimeParseException`) more explicit imo.

File: src/main/java/com/hedvig/botService/chat/OnboardingConversationDevi.kt [hunk 2]
```diff
@@ -1721,6 +1738,14 @@ constructor(
         userContext.completeConversation(this)
     }
 
+    private fun memberIsYoungerThan18(birthDate: LocalDate): Boolean {
+            val dateToday = LocalDate.now()
+
+            val chronoBirthDate = ChronoLocalDate.from(birthDate)
+            val date18YearsAgo = dateToday.minusYears(18)
+            return chronoBirthDate.isAfter(date18YearsAgo)
```
Comment (palmenhq) [inline src/main/java/com/hedvig/botService/chat/OnboardingConversationDevi.kt]: just a thought, maybe this should be `!chronoBirthDAte.isBefore(date18YearsAgo)`? if  today is their birthday

Comment (palmenhq) [reply src/main/java/com/hedvig/botService/chat/OnboardingConversationDevi.kt]: time is so extremely hard to think about though, can we just make a test that tests if you're exactly 18

Comment (AlexSJ1990) [reply src/main/java/com/hedvig/botService/chat/OnboardingConversationDevi.kt]: will add a test for 18 years exactly

**Comment by palmenhq** [comment]
Can we do this for web-onboarding also? I think we're checking it already but at age 12 or something

**Pull request merged**